### ISSUE TITLE
Extra path shown in Pythonpath manager when using the Mac app

### ISSUE THREAD

**Issue by jiayilan** [spyder-ide/spyder#20637]
### What steps will reproduce the problem?

i just installed spyder 5 and when i opened it, the kernerl showed error. i tried to install pip install spyder-kernels==2.2.1, but didn't work

### Traceback
```python-traceback
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/jupyter_client/manager.py", line 85, in wrapper
    self._ready.set_result(None)
asyncio.exceptions.InvalidStateError: invalid state

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "qtconsole/manager.pyc", line 27, in poll
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/jupyter_client/restarter.py", line 143, in poll
    self.kernel_manager.restart_kernel(now=True, newports=newports)
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/jupyter_client/utils.py", line 30, in wrapped
    raise e
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/jupyter_client/utils.py", line 27, in wrapped
    return loop.run_until_complete(future)
  File "nest_asyncio.pyc", line 90, in run_until_complete
  File "asyncio/futures.pyc", line 201, in result
  File "asyncio/tasks.pyc", line 256, in __step
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/jupyter_client/manager.py", line 548, in _async_restart_kernel
    await ensure_async(self.start_kernel(**self._launch_args))
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/jupyter_client/utils.py", line 30, in wrapped
    raise e
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/jupyter_client/utils.py", line 27, in wrapped
    return loop.run_until_complete(future)
  File "nest_asyncio.pyc", line 90, in run_until_complete
  File "asyncio/futures.pyc", line 201, in result
  File "asyncio/tasks.pyc", line 256, in __step
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/jupyter_client/manager.py", line 88, in wrapper
    self._ready.set_exception(e)
asyncio.exceptions.InvalidStateError: invalid state
```

## Versions

* Spyder version: 5.4.2 93124668b (standalone)
* Python version: 3.9.14 64-bit
* Qt version: 5.15.2
* PyQt5 version: 5.15.7
* Operating System: Darwin 22.3.0

### Dependencies

```
# Mandatory:
atomicwrites >=1.2.0                 :  1.4.1 (OK)
chardet >=2.0.0                      :  5.1.0 (OK)
cloudpickle >=0.5.0                  :  2.2.0 (OK)
cookiecutter >=1.6.0                 :  2.1.1 (OK)
diff_match_patch >=20181111          :  20200713 (OK)
intervaltree >=3.0.2                 :  3.1.0 (OK)
IPython >=7.31.1;<9.0.0              :  8.8.0 (OK)
jedi >=0.17.2;<0.19.0                :  0.18.2 (OK)
jellyfish >=0.7                      :  0.9.0 (OK)
jsonschema >=3.2.0                   :  4.17.3 (OK)
keyring >=17.0.0                     :  23.13.1 (OK)
nbconvert >=4.0                      :  7.2.8 (OK)
numpydoc >=0.6.0                     :  1.5.0 (OK)
parso >=0.7.0;<0.9.0                 :  0.8.3 (OK)
pexpect >=4.4.0                      :  4.8.0 (OK)
pickleshare >=0.4                    :  0.7.5 (OK)
psutil >=5.3                         :  5.9.4 (OK)
pygments >=2.0                       :  2.14.0 (OK)
pylint >=2.5.0;<3.0                  :  2.15.10 (OK)
pylint_venv >=2.1.1                  :  None (OK)
pyls_spyder >=0.4.0                  :  0.4.0 (OK)
pylsp >=1.7.1;<1.8.0                 :  1.7.1 (OK)
pylsp_black >=1.2.0                  :  1.2.1 (OK)
qdarkstyle >=3.0.2;<3.1.0            :  3.0.3 (OK)
qstylizer >=0.2.2                    :  0.2.2 (OK)
qtawesome >=1.2.1                    :  1.2.2 (OK)
qtconsole >=5.4.0;<5.5.0             :  5.4.0 (OK)
qtpy >=2.1.0                         :  2.3.0 (OK)
rtree >=0.9.7                        :  1.0.1 (OK)
setuptools >=49.6.0                  :  66.0.0 (OK)
sphinx >=0.6.6                       :  5.1.1 (OK)
spyder_kernels >=2.4.2;<2.5.0        :  2.4.2 (OK)
textdistance >=4.2.0                 :  4.5.0 (OK)
three_merge >=0.1.1                  :  0.1.1 (OK)
watchdog >=0.10.3                    :  2.2.1 (OK)
zmq >=22.1.0                         :  24.0.1 (OK)

# Optional:
cython >=0.21                        :  0.29.33 (OK)
matplotlib >=3.0.0                   :  3.6.3 (OK)
numpy >=1.7                          :  1.24.1 (OK)
pandas >=1.1.1                       :  1.5.2 (OK)
scipy >=0.17.0                       :  1.10.0 (OK)
sympy >=0.7.3                        :  1.11.1 (OK)

# Spyder plugins:
spyder_terminal.terminalplugin 1.2.2 :  1.2.2 (OK)
```

### Environment

<details><summary>Environment</summary>

```

```
</details>

**Comment by jiayilan** [comment]
solved, pip install psutil==5.9.4

**Comment by ccordoba12** [comment]
Hey @jiayilan, thanks for reporting. Did you set an external interpreter in our Preferences which didn't have `psutil`?

That would help us to better understand this problem for other users.

**Comment by ccordoba12** [comment]
@dalthviz, could you check if there's an entry added to our Pythonpath manager in the Mac installer? That could be causing this problem in a similar way to issue #20242 for the Windows installer.

**Comment by dalthviz** [comment]
Checking the `PYTHONPATH manager` on macOS I see an entry for `System PYTHONPATH` related to the resources directory of the installer:

![image](https://user-images.githubusercontent.com/16781833/223879299-03bca91e-84e9-49a6-9150-01d2afd6e918.png)

**Comment by ccordoba12** [comment]
Ok, what happens if you try use another environment with the installer? Do you get the error above?

**Comment by dalthviz** [comment]
Tried a custom environment with the installer but seems like it is working 🤔

**Comment by mrclary** [comment]
@ccordoba12 @dalthviz, I think this issue may already be resolved by #20106 (merged after release 5.4.2).
I've confirmed that the standalone Spyder.app version 5.4.2 provides a System PYTHONPATH with `/Applications/Spyder.app/Contents/Resources`. However, the latest pull request artifact on 5.x (https://github.com/spyder-ide/spyder/pull/20656/checks) does not provide `/Applications/Spyder.app/Contents/Resources`. This makes sense to me since #20106 fixed a lot of issues related to getting the system PYTHONPATH.

**Comment by ccordoba12** [comment]
Ok, thanks for checking that @mrclary. Closing the as fixed by #20106.
